ompi/op/op.h

```diff
@@ -588,15 +573,23 @@ static inline void ompi_op_reduce(ompi_op_t * op, const void *source,
                                           op->o_func.java_data.object);
         return;
     }
-    op->o_func.c_fn(source, target, &count, &dtype);
+    if (0 == (op->o_flags & OMPI_OP_FLAGS_BIGCOUNT)) {
+        op->o_func.c_fn(source, target, &count, &dtype);
```

Comment:
Should [this code](https://github.com/open-mpi/ompi/pull/13015/files#diff-dfd595860dc4d0c9c73114bd8563c0786d063327f6e30955a3e361c3f6098e3dL519) be moved here? We still need to handle larger counts even if the user hasn't provided a big-count callback.

Reply:
I'm not sure I understand the comment.  What is the this code link referring to?  My understanding of the standard is that for user defined reduction functions that if they are expecting to need to handle big count elements they need to define a bigcount aware reduction op.

Reply:
You removed the blob of code that deals with larger-than INT_MAX elements here: https://github.com/open-mpi/ompi/pull/13015/files#diff-dfd595860dc4d0c9c73114bd8563c0786d063327f6e30955a3e361c3f6098e3dL510-L539

Should this be moved into the code path where we deal with user reduction operators that are not big-count ready? Otherwise we might have overflows if the we want to reduce 4M elements with a "legacy" user reduction operator.

Reply:
no that was stuff josh added to handle the fact that the ```ompi_op_reduce``` was already using a size_t full_count arg for some reason before this PR, but our internal op table methods could only handle int.